### PR TITLE
Make non-important failures at least a general softfail

### PR DESCRIPTION
**Pull request by coolo**
This way failed modules have at least *some* significance

### CODE REVIEW

**Comment by coolo** [comment]
for SLES this means +2 softfails for desktop this means +12 softfails, which indicates the under-usage of  important flag for the desktop

**Comment by okurz** [comment]
LGTM but this is a change where others certainly could share their opinion.

@AdamWill maybe? If you are on vacation you missed the chance ;-)

File: lib/OpenQA/Schema/Result/Jobs.pm [hunk 1]
```diff
@@ -676,6 +676,10 @@ sub calculate_result($) {
                 $important_overall = FAILED;
             }
             else {
+                # override important if it's passed
+                if ($important_overall && $important_overall eq PASSED) {
+                    $important_overall = SOFTFAILED;
+                }
                 $overall = FAILED;
```
Comment (aaannz) [inline lib/OpenQA/Schema/Result/Jobs.pm]: Suppose first module is unimportant and fails => `$overall` is FAILED, `$important_overall` is undef
Next module is important and passed => `$overall` if FAILED, `$important_overall` is PASSED
Then test finishes (or the rest of modules pass).
Result is then PASSED even when there is one failed module.

I think this update to result should be at the end of the calculate_result outside of the for cycle.

Comment (coolo) [inline lib/OpenQA/Schema/Result/Jobs.pm]: ok, let me extend the test case by adding another paste block :)

**Comment by coolo** [comment]
Fixed in https://github.com/os-autoinst/openQA/pull/806/commits/d02c6e4aee9f40676477772243acfb0723040746

File: lib/OpenQA/Schema/Result/Jobs.pm [hunk 2]
```diff
@@ -676,10 +677,16 @@ sub calculate_result($) {
...
+                $unimportant_fails = 1;
+                $overall           = FAILED;
             }
         }
     }
+    # override important if it's passed
+    if ($unimportant_fails && $important_overall && $important_overall eq PASSED) {
+        $important_overall = SOFTFAILED;
+    }
+
```
Comment (aaannz) [inline lib/OpenQA/Schema/Result/Jobs.pm]: Fine, but is the extra var necessary?
Isn't this sufficient?
```perl
if ($overall eq FAILED && $important_overall && $important_overall eq PASSED) {
```

**Comment by aaannz** [comment]
Seems to me coveralls has a problem with subtests. Again changes unknown.

File: lib/OpenQA/Schema/Result/Jobs.pm (hunk 2 above)
Comment (coolo) [inline lib/OpenQA/Schema/Result/Jobs.pm]: which begs the question what happens if overall is SOFTFAILED and important_overall is PASSED

Comment (aaannz) [inline lib/OpenQA/Schema/Result/Jobs.pm]: What about?
```perl
if ($important_overall && $important_overall eq PASSED) {
    if ($overall ne PASSED) {
        $important_overall = SOFTFAILED;
  }
}
```

Comment (coolo) [inline lib/OpenQA/Schema/Result/Jobs.pm]: this case wasn't even covered before: ac9246a

**Comment by aaannz** [comment]
:+1:

**Comment by coolo** [comment]
adam will have to post-review this

**Comment by AdamWill** [comment]
sorry, I was in transit. I haven't looked into the impact of this on us yet. It may affect https://phab.qadevel.cloud.fedoraproject.org/D932 .

**Comment by coolo** [comment]
you will in general have to review all your tools that acted on passed/failed as you can have softfails now in the API (you asked for it :)

**Pull request merged**